doc/man3/PKCS12_newpass.pod

```diff
@@ -17,6 +17,16 @@ PKCS12_newpass() changes the password of a PKCS12 structure.
 B<p12> is a pointer to a PKCS12 structure. B<oldpass> is the existing password
 and B<newpass> is the new password.
 
+=head1 NOTES
+
+Each of B<oldpass> and B<newpass> is independently interpreted as a string in
+the UTF-8 encoding. If it is not valid UTF-8, it is assumed to be ISO9859-1
+instead.
+
+In particular, this means that passwords obtained through the default
+UI_OpenSSL() method, which are in the locale character set, should not be
+used without potentially being converted to UTF-8 first.
+
```

Comment:
To be clear: the "potentially" is a shorthand for something along the lines of...

A password string returned by the UI should be converted from the locale character set to UTF-8. Such a conversion may be skipped if:
 • the locale character set *is* UTF-8, or
 • the password entered happens to contain only characters for which the locale character encoding is compatible with UTF-8 (i.e. contains only 7-bit ASCII characters, for most non-EBCDIC locales). or
 • the password entered happens to contain only characters for which the locale character encoding is compatible with ISO8859-1, *and* the resulting byte sequence cannot be interpreted as valid UTF-8.

On the whole, I was happier not writing that, and sticking with *"potentially"* :)